src/pybind/mgr/cephadm/module.py

```diff
@@ -1659,6 +1685,11 @@ def remove_daemons(self, names, force):
             for name in names:
                 if name in dm:
                     args.append((name, host, force))
+                    # TODO: bail out for OSDs when https://github.com/ceph/ceph/pull/32983 is merged
+                    if dm[name].daemon_type == 'osd':
+                        continue
+                    self._remove_key_from_store(dm[name].daemon_type,
+                                                dm[name].service_name(without_service_type=True))
```

Comment:
daemons != services.. I don't think this should be there?

Reply:
The're both querying from the same source -> `self.cache.daemons` so I'd assume they return the same type(DaemonDescription).

Reply:
Okay, but imagine, say, that you have a spec that says "3 mds.foo daemons".  And the users goes and looks at 'orch ps', sees three of them, and decides they don't want one of them.  If they do 'ceph orch daemon rm mds.foo.abcdef', the above code will delete the entire service, which will orphan the other two daemons.

In kubernetes, the analog is a ReplicaSet of 3 pods, and you kill a pod, and you expect the operator/controller to immediate recreate a new one.  IMO we should mimic that behavior.